### PR TITLE
adds lorenz96 model

### PR DESCRIPTION
**Pull request by samgdotson**
This pull request adds a Lorenz-96 model for validating the Echo State Network implementation.

### CODE REVIEW

File: lorenz96-example.ipynb [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
...bz7aitqcxLLtv2BeOnLXsH69nLCtCCGiaxm3YFEjD6v7vBobl4atWqwDY3GsrDMvkO7U1l2E5uoJaWTZN09OZq25IpVKYmppiWB6CYV84VioVNBoNLC0tDeX70+WpGct+9Cfg+CgKKvW8CGpYrtVqaLfbfi8lsqrVKpLJJGZmZvxeSqAxLJPvLMtCIpHA2NiY30uhIcnlckin04FqSuU4DizLClVzL6VYLDIsD8Gww7I6v8rqsvdM00QikfDlpqymadB1HY7jeP7YRJvRdR25XC6QzZ3UcRgV6Ml98/PzmJmZCdSM7SBiWCbf2baNqampUJ0bpe0RQgRu1rJt23AcJ3TbsAHOWh4WdeE4Ojo6lO/PZk/+MQwD+XweiYT3lz2apqHb7fKinwInyEeROD5quFRzr9nZWb+XEnhDedcQQjCEU884YzkeghaW1VrCtg0bWL2IMAyDlSqXDWtslFIqlSCEYFj2gWEYvu0i4fgoCqphdf93gwrLfN4Mx5kzZ9BsNnleuQc9hVohxI+FEA+u+/ecEOLTQojrNvjaXwbAAwbUM8uy2Ak7BvL5fKC2YfvV8McNxWJxbRs5uWfYF46pVArFYpHbsH1gmqZvz3XuKKCgMgwjkOeVgdU+NqlUiruohoTNvXrXawX4WgAT6/59FMDHAVzp+ooodlhZjgdVWZZS+r0UAP6NknGDuujn9jT3qBnLwxobpXB8lPeklDBN07fn+tTUFNLpNH/vFCjLy8tYXl4O7A1jIQSKxSLf54akWq1iZGQEO3fu9HspgTfIdmkeMKWBOY6DxcVFhuUYKBQKcBwH9Xrd76UAWA3L6XQauVzO76VsGxufuK/RaKDVag29yqJpGs6cOROYm0ZxsLi4iE6n41soEEKgXC5zOykFSpDHRimlUonPmyGpVqvYvXu3L30cwoY/IfKVbduQUjIsx0DQxkepSlMYG8upi35uT3PPsGcsK5qmodlsBuamURz4OTZKYVimoAny2CilXC7zfW4Iut0uFhYWuAW7RwzL5Ct15pJhOfpUI62gnFv2c1vmoNLpNCYnJ3kR4SL1sxz2NmyeX/VeEI5caJqGs2fPckcBBYZXNwgHUSwW0Wg00Gg0/F5KpJw6dQrtdpthuUcMy+QrhuX4CFpl2c/uuG7grGV3eXXhWKlUAHDWspeC0MxP0zS02+3A3CwkMgwDqVQKExM
```
Comment (abachma2) [inline lorenz96-example.ipynb:617]: Is this meant to be here? The last cell just has an s in it.

File: lorenz96.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    Code snippet adapted from a Wikipedia example.
+    Here the differential equation is described for
+    use with the ``odeint`` function.
+
+    Parameters:
+    -----------
+    x : The independent variable
+    t : The time of the simulation
+    N : integer
+        The number of variables in the system
```
Comment (abachma2) [inline lorenz96.py:19]: N >= 4 is a constraint for generate_L96. Is it also a constraint for this function?

**Review by abachma2 — COMMENTED** [review]
Overall, looks good. The predictions seem to be pretty strong. Just a few questions on small things.

File: lorenz96.py (hunk 2 above)
Comment (samgdotson) [reply lorenz96.py:19]: It's not really a constraint. The model is still defined for N < 4, it's just less interesting. Fixed the doc string

File: lorenz96-example.ipynb (hunk 1 above)
Comment (samgdotson) [reply lorenz96-example.ipynb]: Fixed

**Review by abachma2 — APPROVED** [review]
Thanks for making the changes. I'll go ahead and merge it now

**Pull request merged**